File: 05-while-loop/exercises-submitted-evaluated/08.graduation.js
function graduation(input) {
    let name = String(input[0]);
    let year = 1;
    let yearlyGrade = Number(input[year]);
    let sumGrades = 0;
    let avgGrade = 0;
    let failures = 0;

    while (year <= 12) {
        if (yearlyGrade < 4) {
            failures += 1;
        }
        if (failures > 1) {
            console.log(`${name} has been excluded at ${year - 1} grade`);
            break;
        }
        sumGrades += yearlyGrade;
        avgGrade = sumGrades / year;
        if (year === 12) {
            console.log(`${name} graduated. Average grade: ${avgGrade.toFixed(2)}`);
            break;
        }
        year++;
        yearlyGrade = Number(input[year]);
    }

}

graduation(["Gosho", 
"5",
"5.5",
"6",
"5.43",
"5.5",
"6",
"5.55",
"5",
"6",
"6",
"5.43",
"5"]);
graduation(["Mimi", 
"5",
"6",
"5",
"6",
"5",
"6",
"6",
"2",
"3"]);

File: softuni-python-basics/05_while_loop/08_graduation.py
student_name = input()
year = 0
fail_counter = 0
total_sum = 0
gpa = 0

while year <= 12:
    yearly_grade = float(input())
    year += 1
    if yearly_grade < 4:
        fail_counter += 1
        if fail_counter == 2:
            print(f"{student_name} has been excluded at {year} grade")
            break
        year -= 1
    else:
        total_sum += yearly_grade
    if year == 12:
        gpa = total_sum / 12
        print(f"{student_name} graduated. Average grade: {gpa:.2f}")
        break
